designer/client/src/components/modals/DeployWithParametersDialog.tsx

```diff
@@ -0,0 +1,137 @@
+    // TODO: get rid of meta
+    const {
+        meta: { action, displayWarnings },
+    } = props.data;
+    const processName = useSelector(getProcessName);
+
+    const activityParameters = useSelector(getActionParameters);
+    const activityNodeParameters = activityParameters["DEPLOY"] || ([] as ActionNodeParameters[]);
+    const initialValues = useMemo(() => initialNodesData(activityNodeParameters), [activityNodeParameters]);
+    const [values, setValues] = useState(initialValues);
+
+    const [comment, setComment] = useState("");
+    const [validationError, setValidationError] = useState("");
+    const featureSettings = useSelector(getFeatureSettings);
+    const deploymentCommentSettings = featureSettings.deploymentCommentSettings;
+
+    const dispatch = useDispatch();
+
+    const confirmAction = useCallback(async () => {
+        try {
+            await action(processName, comment, values);
+            props.close();
+        } catch (error) {
+            setValidationError(error?.response?.data);
+        }
+    }, [action, comment, dispatch, processName, props]);
+
+    const { t } = useTranslation();
+    const buttons: WindowButtonProps[] = useMemo(
+        () => [
+            { title: t("dialog.button.cancel", "Cancel"), action: () => props.close(), classname: LoadingButtonTypes.secondaryButton },
+            { title: t("dialog.button.ok", "Ok"), action: () => confirmAction() },
+        ],
+        [confirmAction, props, t],
+    );
+
+    return (
+        <PromptContent {...props} buttons={buttons}>
+            <div className={cx("modalContentDark")}>
+                <Typography variant={"h3"}>{props.data.title}</Typography>
+                {displayWarnings && <ProcessDialogWarnings />}
+                <CommentInput
+                    onChange={(e) => setComment(e.target.value)}
+                    value={comment}
+                    defaultValue={deploymentCommentSettings?.exampleComment}
+                    className={cx(
+                        css({
+                            minWidth: 600,
+                            minHeight: 80,
+                        }),
+                    )}
+                    autoFocus
+                />
+                <FormHelperText title={validationError} error>
+                    {validationError}
+                </FormHelperText>
+                {activityNodeParameters && activityNodeParameters.length > 0 ? (
+                    <Typography
+                        sx={(theme) => ({
+                            color: theme.palette.primary.main,
+                            pt: "1em",
+                            textTransform: "uppercase",
+                            textDecoration: "none",
+                        })}
+                    >
+                        Advanced parameters
+                    </Typography>
+                ) : null}
+                {activityNodeParameters.map((anp: ActionNodeParameters) => (
```

Comment:
This acronym was a little confusing to me. I needed to go back to activityNodeParameters to understand what was going on. WDYT about renaming it to activityNodeParameter?

Reply:
Fixed.